titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/DefaultKubeJobManagementReconciler.java

```diff
@@ -167,13 +205,15 @@ private void reconcile() {
         List<Task> tasks = v3JobOperations.getTasks();
 
         Map<String, V1Node> nodesById = nodes.stream().collect(Collectors.toMap(
-                node -> node.getMetadata().getName(),
+                node -> KubeUtil.getMetadataName(node.getMetadata()),
                 Function.identity()
         ));
         Map<String, Task> currentTasks = tasks.stream().collect(Collectors.toMap(Task::getId, Function.identity()));
-        Set<String> currentPodNames = pods.stream().map(p -> p.getMetadata().getName()).collect(Collectors.toSet());
+        Set<String> currentPodNames = pods.stream().map(p -> KubeUtil.getMetadataName(p.getMetadata())).collect(Collectors.toSet());
```

Comment:
same here, filter out the empty string to void trying to call delete with an empty name later